Documentation for midi_note_on and midi_note_off examples  improved

amruth-acharya opened the pull request:
Hi, The documentation for midi_note_on and midi_note_off examples are improved.  The comments  are updated for midi_note_on and midi_note_off.
 
Fixes #2915

amruth-acharya committed:
Merge pull request #1 from amruth-acharya/docs-2

Fixed Extempore link in contributors.md

amruth-acharya committed:
Merge branch 'sonic-pi-net:dev' into main

amruth-acharya committed:
Merge branch 'sonic-pi-net:dev' into main

amruth-acharya committed:
Merge branch 'sonic-pi-net:dev' into main

amruth-acharya committed:
Merge branch 'sonic-pi-net:dev' into main

amruth-acharya committed:
Documentation for midi_note_on and midi_note_off examples  improved

Hi, The documentation for midi_note_on and midi_note_off examples are improved.  The comments  are updated for midi_note_on and midi_note_off.
 
Fixes #2915

amruth-acharya commented:
@ethancrawford what should i do?

ethancrawford commented on app/server/ruby/lib/sonicpi/lang/midi.rb:
```diff
@@ -377,15 +377,15 @@ def midi_note_on(*args)
-        "midi_note_on :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note :e3 on to on all channels on ports named \"foo\" and \"bar\"",
-        "midi_note_on :e3, channel: 1, port: \"foo\" #=> Send MIDI note :e3 on only on channel 1 on port \"foo\""
+        "midi_note_on :e3, channel: [1, 3, 5] #=> Send MIDI note on for note :e3 to channels 1, 3, 5 on all connected ports",
+        "midi_note_on :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note on for note :e3 to all channels on ports named "foo" and "bar",
```
```suggestion
        "midi_note_on :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note on for note :e3 to all channels on ports named \"foo\" and \"bar\"",
```

ethancrawford commented on app/server/ruby/lib/sonicpi/lang/midi.rb:
```diff
@@ -377,15 +377,15 @@ def midi_note_on(*args)
-        "midi_note_on :e3, channel: 1, port: \"foo\" #=> Send MIDI note :e3 on only on channel 1 on port \"foo\""
+        "midi_note_on :e3, channel: [1, 3, 5] #=> Send MIDI note on for note :e3 to channels 1, 3, 5 on all connected ports",
+        "midi_note_on :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note on for note :e3 to all channels on ports named "foo" and "bar",
+        "midi_note_on :e3, channel: 1, port: \"foo\" #=> Send MIDI note on for note :e3 only on channel 1 on port "foo"
```
```suggestion
        "midi_note_on :e3, channel: 1, port: \"foo\" #=> Send MIDI note on for note :e3 only on channel 1 on port \"foo\""
```

ethancrawford commented on app/server/ruby/lib/sonicpi/lang/midi.rb:
```diff
@@ -448,15 +448,15 @@ def midi_note_off(*args)
+        "midi_note_off :e3, 12, channel: 3  #=> Sends MIDI note off for note :e3 with velocity 12 to channel 3",
+        "midi_note_off :e3, velocity: 100 #=> Sends MIDI note off for note :e3 with release velocity 100",
+        "midi_note_off :e3, vel_f: 0.8 #=> Scales release velocity 0.8 to MIDI value 102 and sends MIDI note off for note :e3 with release velocity 102",
+        "midi_note_off 60.3, 50.5 #=>Rounds params up or down to the nearest whole number and sends MIDI note off for note 60 with release velocity 51",
```
```suggestion
        "midi_note_off 60.3, 50.5 #=> Rounds params up or down to the nearest whole number and sends MIDI note off for note 60 with release velocity 51",
```

ethancrawford commented on app/server/ruby/lib/sonicpi/lang/midi.rb:
```diff
@@ -448,15 +448,15 @@ def midi_note_off(*args)
+        "midi_note_off :e3, vel_f: 0.8 #=> Scales release velocity 0.8 to MIDI value 102 and sends MIDI note off for note :e3 with release velocity 102",
+        "midi_note_off 60.3, 50.5 #=>Rounds params up or down to the nearest whole number and sends MIDI note off for note 60 with release velocity 51",
+        "midi_note_off :e3, channel: [1, 3, 5] #=> Send MIDI note off for note :e3 to channels 1, 3, 5 on all connected ports",
+        "midi_note_off :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note off for note :e3 to all channels on ports named "foo" and "bar"",
```
```suggestion
        "midi_note_off :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note off for note :e3 to all channels on ports named \"foo\" and \"bar\"",
```

ethancrawford commented on app/server/ruby/lib/sonicpi/lang/midi.rb:
```diff
@@ -448,15 +448,15 @@ def midi_note_off(*args)
+        "midi_note_off 60.3, 50.5 #=>Rounds params up or down to the nearest whole number and sends MIDI note off for note 60 with release velocity 51",
+        "midi_note_off :e3, channel: [1, 3, 5] #=> Send MIDI note off for note :e3 to channels 1, 3, 5 on all connected ports",
+        "midi_note_off :e3, port: [\"foo\", \"bar\"] #=> Send MIDI note off for note :e3 to all channels on ports named "foo" and "bar"",
+        "midi_note_off :e3, channel: 1, port: \"foo\" #=> Send MIDI note off for note :e3 only on channel 1 on port "foo""
```
```suggestion
        "midi_note_off :e3, channel: 1, port: \"foo\" #=> Send MIDI note off for note :e3 only on channel 1 on port \"foo\""
```

ethancrawford reviewed (commented):
A few small adjustments needed. (Remember, I mentioned that all of the special characters _including_ escape characters such as the backslash `\`, as well as all of the quotes `"` are still needed 🙂 - the reason is that Ruby is very particular about how strings are formatted when you want to have nested quotes inside a string.

A string itself is created by wrapping text in quotes: `"Hello world"`. But if you want strings to have quotes _inside them_ as well, you need to escape them with `\` like this: `"A string that says \"Hello world\""`. That tells Ruby that the inside quotes are part of the text of your string, not markers for where the string starts or ends. Otherwise, if these inner quotes are not escaped, the string will be broken and the surrounding code won't work (notice how the build checks at the bottom of this PR page failed? that's why 🙂)  

You can commit the suggestions I have added here with the buttons that are underneath them - they will fix the string formatting problems 👍

amruth-acharya committed:
Update app/server/ruby/lib/sonicpi/lang/midi.rb

Co-authored-by: ethancrawford <ethan_jc@hotmail.com>

amruth-acharya committed:
Update app/server/ruby/lib/sonicpi/lang/midi.rb

Co-authored-by: ethancrawford <ethan_jc@hotmail.com>

amruth-acharya committed:
Update app/server/ruby/lib/sonicpi/lang/midi.rb

Co-authored-by: ethancrawford <ethan_jc@hotmail.com>

amruth-acharya commented:
@ethancrawford  thanks for the help .Quick question if i want edit a pullrequest (just as you asked changes in pr)  how to update the existing pullrequest?

ethancrawford commented:
There are a few ways.
If people add a suggestion in a review comment (like I did above), GitHub is smart enough to show you what the suggested change will look like against your code, and allows you to commit the suggestion into your PR with a button right here on GitHub, as you seem to have already done for some of my suggestions above. It's quick and easy. (Note there are still two suggestions that haven't been committed to the PR yet).

You can also make changes to the PR by just editing your fork of the repo on your local machine, provided you're on the same branch that the PR was created from, (in this case, 'amruth-midi'), just like you were doing when you started making your changes for this PR to begin with. Then you push your changes to the online copy of your fork on GitHub with the `git push` command. (There's less need to do this if the changes you need to make to your PR are only changes mentioned in code change suggestions, because they can be made from GitHub directly).

amruth-acharya commented:
> There are a few ways. If people add a suggestion in a review comment (like I did above), GitHub is smart enough to show you what the suggested change will look like against your code, and allows you to commit the suggestion into your PR with a button right here on GitHub, as you seem to have already done for some of my suggestions above. It's quick and easy. (Note there are still two suggestions that haven't been committed to the PR yet).
> 
> You can also make changes to the PR by just editing your fork of the repo on your local machine, provided you're on the same branch that the PR was created from, (in this case, 'amruth-midi'), just like you were doing when you started making your changes for this PR to begin with. Then you push your changes to the online copy of your fork on GitHub with the `git push` command. (There's less need to do this if the changes you need to make to your PR are only changes mentioned in code change suggestions, because they can be made from GitHub directly).

which two PR are remaining? (Github  isn't my thing excuse my noobness)

amruth-acharya committed:
Documentation for midi_note_on and midi_note_off examples improved

Documentation for midi_note_on and midi_note_off examples improved and In comments the quotes were missing and they are fixed.

amruth-acharya commented:
@ethancrawford If i want  to delete commits  in future how do i do it, can the commits in remote can be deleted? I searched on web, but they look confusing (sry for asking too many questions ) 😅

ethancrawford reviewed (approved):
Looks good to me 🙂

amruth-acharya commented:
> Looks good to me 🙂
😅

ethancrawford commented:
@amruth-acharya - it is possible to do something similar to 'deleting' a commit, but not always desired. (It's something you want to avoid if you are working on a branch that other people have also been working on for example).

On GitHub here, if the repository is set up by the owner in the right way, GitHub allows us to merge all your changes into the repo as one single git commit (by doing a 'squash and merge'), even if you added several commits in your PR. Sam's repo is not set up this way though, so that option is out for us.

The other ways to 'delete' a commit here on GitHub would be to use commands to change the state of your repo on your local machine, and then push those changes up to GitHub, so that the PR is updated with your changes.
There are several ways to use git to undo changes to your code, (including doing something similar to 'deleting' commits) but there are always things to consider when doing this - which method you use will depend on whether any changes you make to undo something will potentially break other people's copies of your code, if they have copies of it on their own devices.

For example, one way to change your PR here so that it was instead one single commit instead of 10, is to use the `git reset --hard` command on you local machine to tell git to go back to the point in your repo before you made all the changes, and then you could make them all as a single commit, and then force push the changes back up to GitHub.
However, this would _only_ be a good idea if you knew other people had not been using the branch that you had made the changes on - if they had, then their copies would now have commits that no longer existed.

It's worth reading up about the ways to undo changes using git:
See https://opensource.com/article/18/6/git-reset-revert-rebase-commands for a good introduction.

amruth-acharya commented:
thanks! trying to get better thorough contributing 😅

samaaron commented:
Thanks so much for this :-)
